pkg/operator/controller/dns/controller.go

```diff
@@ -595,13 +595,20 @@ func (r *reconciler) createDNSProvider(dnsConfig *configv1.DNS, platformStatus *
 	case configv1.PowerVSPlatformType:
 		//Power VS platform will use the ibm dns implementation
 		var err error
-		cisInstanceCRN := platformStatus.PowerVS.CISInstanceCRN
-		if cisInstanceCRN == "" {
-			return nil, fmt.Errorf("missing cis instance crn")
-		}
-		dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, cisInstanceCRN, userAgent, true)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create IBM DNS manager: %v", err)
+		if platformStatus.PowerVS.CISInstanceCRN != "" {
+			dnsProvider, err = getIbmDNSProvider(dnsConfig, creds, platformStatus.PowerVS.CISInstanceCRN, userAgent, true)
+			if err != nil {
+				return nil, err
+			}
+		} else if platformStatus.PowerVS.DNSInstanceCRN != "" {
+			matches := ibm.IBMResourceCRNRegexp.FindStringSubmatch(platformStatus.PowerVS.DNSInstanceCRN)
```

Comment:
Wondering if we could get rid of the code duplication by moving the common parts to `getIbmDNSProvider` as provider creation seems to be the same for both IBMCloud VPC and PowerVS.

Reply:
Thank you, I have update the code accordingly, Please take a look.

Reply:
Looks good to me.